### ISSUE TITLE
Configured slots don't sync to client when connecting to a dedi server

### ISSUE THREAD

**Issue by LemmaEOF** [copygirl/Wearables#4]
Resource reload listeners don't automatically sync from dedicated servers to connected clients. Because of that, clients don't get GUI access to the configured slots when connected to a server. There are a couple ideas I have for how packets could be synced:
1. just send the raw text of the JSON over and have the client deserialize it (probably really inefficient, but the deserialize code is already there)
2. write a PacketByteBuf serializer/deserializer for the configuration and send that (we'd need to think up an efficient spec to send over only the info needed, but it'd be much more performant)

This packet would also need to be sent whenever a player joins, not just on resource reload.

**Comment by copygirl** [comment]
Alternatively, we could skip sending most of the data and just send as necessary. For example, the client doesn't need to know which slots are available on an entity, it just needs to know when a container is opened that allows it to access these Wearables slots.

If we do this right, it would probably also make unlockable slots and such that are controlled by the server easier, which is a feature I'd like to support some time in the future.

So basically, when player opens a container (and initially, for the player's own inventory), send which slots are available specifically through that container. That way we can avoid syncing most config data, and slots are controlled by the server. The only configuration we might need to send is region positioning.

**Comment by LemmaEOF** [comment]
That sounds good!

**Comment by falkreon** [comment]
I think that could work - if it's in the packetbytebuf data fired down with the open-container call, it squeezes in just before container sync where changing the number of slots is Very Bad

**Comment by copygirl** [comment]
Should be fixed now that we're straight up using container slots now, which - I believe - sync for free.
